packages/api/amplify_api/lib/src/amplify_authorization_rest_client.dart

```diff
@@ -30,30 +39,42 @@ class AmplifyAuthorizationRestClient extends AWSBaseHttpClient {
 
   /// The authorization mode to use for requests with this client.
   ///
-  /// If provided, will override the [authorizationType] of [endpointConfig].
+  /// If provided, will override the authorizationType of [endpointConfig].
   final APIAuthorizationType? authorizationMode;
 
-  /// Provide an [AWSApiConfig] which will determine how requests from this
-  /// client are authorized.
-  AmplifyAuthorizationRestClient({
-    required this.endpointConfig,
-    required this.authProviderRepo,
-    this.authorizationMode,
-    AWSHttpClient? baseClient,
-  }) : baseClient = baseClient ?? AWSHttpClient();
-
   @override
   final AWSHttpClient baseClient;
 
   /// Implementation of [transformRequest] that authorizes any request without "Authorization"
-  /// or "X-Api-Key" header already set.
+  /// or "X-Api-Key" header already set and enforces HTTPS.
   @override
   Future<AWSBaseHttpRequest> transformRequest(AWSBaseHttpRequest request) {
+    if (request.scheme != 'https') {
+      throw const ApiException('Non-HTTPS requests not supported.');
+    }
     return authorizeHttpRequest(
       request,
       endpointConfig: endpointConfig,
       authProviderRepo: authProviderRepo,
       authorizationMode: authorizationMode,
     );
   }
+
+  @override
+  Future<AWSBaseHttpResponse> transformResponse(
+    AWSBaseHttpResponse response,
+  ) async {
+    // For REST endpoints, throw [RestException] on non-successful responses.
+    if (endpointConfig.endpointType == EndpointType.rest &&
+        (response.statusCode < 200 || response.statusCode >= 300)) {
```

Comment:
Are we throwing for 3xx?

Reply:
Yes. 1) to be consistent w current behavior 2) though I guess you could make the case for following redirects that's not a change to behavior that was in API review.

Reply:
Got it. Did not know that was the current behavior.